packages/fxa-auth-server/lib/account-delete.ts

```diff
@@ -197,30 +180,38 @@ export class AccountDeleteManager {
    * Delete the account from the FxA database, OAuth database, pushbox database, and
    * cancel any active subscriptions.
    *
-   * @param uid
+   * @param accountRecord Account
+   * @param options AccountDeleteOptions optional object with an optional `notify` function that will be called after the account's removed from MySQL.
    */
-  public async deleteAccount(uid: string) {
-    const accountRecord = await this.fxaDb.account(uid);
+  public async deleteAccount(uid: string, options?: AccountDeleteOptions) {
+    await this.deleteAccountFromDb(uid, options);
+    await this.deleteOAuthTokens(uid);
+    // see comment in the function on why we are not awaiting
+    this.deletePushboxRecords(uid);
 
-    await this.deleteSubscriptions(accountRecord);
-
-    await this.deleteAccountDb(accountRecord);
-    this.log.info('accountDeleted.byRequest', accountRecord);
-
-    await this.deleteOAuthDb(accountRecord);
-
-    await this.deletePushboxRecord(accountRecord);
+    await this.deleteSubscriptions(uid);
+    await this.deleteFirestoreCustomer(uid);
+  }
 
-    // TODO: Update to send notification to all devices
+  public async cleanupAccount(uid: string) {
+    await this.deleteSubscriptions(uid);
+    await this.deleteFirestoreCustomer(uid);
+    await this.deleteOAuthTokens(uid);
   }
 
   /**
    * Delete the account from the FxA database.
    *
    * @param accountRecord
    */
-  public async deleteAccountDb(accountRecord: Account) {
-    await this.fxaDb.deleteAccount(accountRecord);
+  public async deleteAccountFromDb(
+    uid: string,
+    options?: AccountDeleteOptions
+  ) {
+    await this.fxaDb.deleteAccount({ uid });
+    if (options?.notify) {
```

Comment:
In the previous version, we only notified after all account records were deleted, not just from FxA db. Was there a reason to notify only after the FxA db delete?

Reply:
The FxA db is the source of truth, in that once the user's account record has been deleted they can no longer sign in or do anything.  Delaying this means that
 - RPs and devices may never receive the notification if any of the other delete calls throws an error, or
 - RPs and devices receive the notification after some delay, during which the user cannot interact with FxA

I'm comfortable with delaying since it's been working fine 🤞.  But I don't see the advantage of it.

Reply:
I think it is probably ok to keep what you have. Maybe in some edge case an RP recieves a delete event and tries to interact with tokens/subscriptions, but they should be able to recover from the error.

Reply:
> Actually we do need to do the customer delete first anyway since that rely on the account record still existing in the db.

Sorry about the bad info, that is _not_ the case.